File: document-ui/src/main/java/dev/sayaya/handbook/client/usecase/TypeRepository.java
package dev.sayaya.handbook.client.usecase;

import dev.sayaya.handbook.client.domain.TypeInfo;
import dev.sayaya.rx.Observable;

/** 타입 목록 조회 포트 인터페이스. 컬럼 정의를 위해 사용한다. */
public interface TypeRepository {
    Observable<TypeInfo[]> list();
}

File: validator/src/main/kotlin/dev/sayaya/handbook/usecase/TypeRepository.kt
package dev.sayaya.handbook.usecase

import dev.sayaya.handbook.domain.Type
import reactor.core.publisher.Mono
import java.time.Instant
import java.util.UUID

interface TypeRepository {
    fun cache(workspace: UUID, type: Type): Mono<Void>
    fun find(workspace: UUID, id: String, effectDateTime: Instant, expireDateTime: Instant): Mono<Type>
}